diff -u cryptlib.h-orig cryptlib.h
--- cryptlib.h-orig	Tue Nov  6 05:52:16 2001
+++ cryptlib.h	Mon Apr 22 19:01:29 2002
@@ -727,7 +727,7 @@
 	*/
 	virtual void GenerateKeyPair(RandomNumberGenerator &rng, byte *privateKey, byte *publicKey) const =0;
 	//! derive agreed value from your private key and couterparty's public key, return false in case of failure
+	//! Note: If you have previously validated the public key, use validateOtherPublicKey=false to save time.
-	/*! Note: If you have previously validated the public key, use validateOtherPublicKey=false to save time.
 	/*! Preconditions:
 			\begin{itemize}
 			\item size of agreedValue == AgreedValueLength()
